### PR TITLE
Set TCP keepalive for mongo nodes

### PR DESCRIPTION
**Pull request by lawrencegripper**
Update the mongo install script to set TCP Keep alives for the nodes and add a `readme` on how to update the binary data in the template.

### CODE REVIEW

File: opencga-app/app/scripts/docker/build-all.sh [hunk 1]
```diff
@@ -113,9 +113,9 @@ if [ ! -d "${buildPath}" ]; then
     docker run -it --rm \
     -v "$PWD":/src \
     -v "$HOME/.m2":/root/.m2 \
-    -w /src maven:3.6-jdk-8 \
-    mvn -T 1C install \
-    -DskipTests -Dstorage-hadoop -Popencga-storage-hadoop-deps -P"${HADOOP_FLAVOR}" -DOPENCGA.STORAGE.DEFAULT_ENGINE=hadoop -Dopencga.war.name=opencga
+    -w /src maven:3.5.3-jdk-8 \
+    mvn clean install -U \
```
Comment (jjcollinge) [inline opencga-app/app/scripts/docker/build-all.sh]: @j-coll I added -U only due the issue I ran into with biodata not compiling - is it safe/wise to leave this in?

Comment (lawrencegripper) [reply opencga-app/app/scripts/docker/build-all.sh]: @j-coll

**Comment by lawrencegripper** [comment]
@jjcollinge looks like I've caused a bug there seeing the mongo not starting on deployment. Going to try and resolve 

```
opencgaadmin@mongo-vm0:~$ cloud-init status --long
status: error
time: Tue, 19 Feb 2019 11:11:40 +0000
detail:
('scripts-user', RuntimeError('Runparts: 1 failures in 1 attempted commands',))
```

```
 sudo docker logs 18b0f5ad9149
Warning: Permanently added 'cgahbaseo7aporm45bnfq-ssh.azurehdinsight.net,52.166.22.211' (ECDSA) to the list of known hosts.
Authorized uses only. All activity may be monitored and reported.
Fetching HDInsight configuration
Warning: Permanently added 'cgahbaseo7aporm45bnfq-ssh.azurehdinsight.net,52.166.22.211' (ECDSA) to the list of known hosts.
Authorized uses only. All activity may be monitored and reported.
Warning: Permanently added 'cgahbaseo7aporm45bnfq-ssh.azurehdinsight.net,52.166.22.211' (ECDSA) to the list of known hosts.
Authorized uses only. All activity may be monitored and reported.
Warning: Permanently added 'cgahbaseo7aporm45bnfq-ssh.azurehdinsight.net,52.166.22.211' (ECDSA) to the list of known hosts.
Authorized uses only. All activity may be monitored and reported.
Initialising configs
Initialising volume
Installing catalog
com.mongodb.MongoTimeoutException: Timed out after 30000 ms while waiting to connect. Client view of cluster state is {type=UNKNOWN, servers=[{address=cgamongolfhhadzwkhrmo0.westeurope.cloudapp.azure.com:27017, type=UNKNOWN, state=CONNECTING, exception={com.mongodb.MongoSocketOpenException: Exception opening socket}, caused by {java.net.ConnectException: Connection refused (Connection refused)}}]
	at com.mongodb.connection.BaseCluster.getDescription(BaseCluster.java:167)
	at com.mongodb.client.internal.MongoClientDelegate.getConnectedClusterDescription(MongoClientDelegate.java:106)
	at com.mongodb.client.internal.MongoClientDelegate.createClientSession(MongoClientDelegate.java:77)
	at com.mongodb.client.internal.MongoClientDelegate$DelegateOperationExecutor.getClientSession(MongoClientDelegate.java:185)
	at com.mongodb.client.internal.MongoClientDelegate$DelegateOperationExecutor.execute(MongoClientDelegate.java:141)
	at com.mongodb.client.internal.MongoCollectionImpl.executeCount(MongoCollectionImpl.java:197)
	at com.mongodb.client.internal.MongoCollectionImpl.count(MongoCollectionImpl.java:177)
	at com.mongodb.client.internal.MongoCollectionImpl.count(MongoCollectionImpl.java:172)
	at org.opencb.commons.datastore.mongodb.MongoDBNativeQuery.count(MongoDBNativeQuery.java:62)
	at org.opencb.commons.datastore.mongodb.MongoDBCollection.count(MongoDBCollection.java:129)
	at org.opencb.opencga.catalog.db.mongodb.MongoDBAdaptorFactory.isCatalogDBReady(MongoDBAdaptorFactory.java:244)
	at org.opencb.opencga.catalog.managers.CatalogManager.existsCatalogDB(CatalogManager.java:161)
	at org.opencb.opencga.app.cli.admin.executors.CatalogCommandExecutor.install(CatalogCommandExecutor.java:153)
	at org.opencb.opencga.app.cli.admin.executors.CatalogCommandExecutor.execute(CatalogCommandExecutor.java:71)
	at org.opencb.opencga.app.cli.admin.AdminMain.main(AdminMain.java:129)
cp: cannot stat '/opt/opencga/sessions/*': No such file or directory
```

**Comment by marrobi** [comment]
I've seen that previously - #1151

**Comment by marrobi** [comment]
Might not be the same cause, but had to redeploy and mongo came up.

**Comment by lawrencegripper** [comment]
Closing this resolves #1076 #1156 and #1113 :)

**Pull request merged**